File: Binary Search/RecursiveBinarySearch.java
import java.util.Scanner;

public class RecursiveBinarySearch {

    public static void main(String[] args) {

        int size, searchingItem;
        System.out.print("Enter array size : ");
        Scanner input = new Scanner(System.in);
        size = input.nextInt();
        int[] numbers = new int[size];

        System.out.print("Enter sorted array elements : ");
        for (int i = 0; i < size; i++) {
            numbers[i] = input.nextInt();
        }
        System.out.printf("Enter the number you wanna search: ");
        searchingItem = input.nextInt();
        input.close();

        int index = binarySearch(numbers, 0, numbers.length - 1, searchingItem);
        if (index == -1) {
            System.out.println("Item not Found");
        } else {
            System.out.println("Item found at index : " + index);
        }
    }

    public static int binarySearch(int num[], int left, int right, int search) {

        int mid = (left + right) / 2;

        if (left <= right) {

            if (num[mid] == search) {
                return mid;
            } else if (num[mid] < search) {
                return binarySearch(num, mid + 1, right, search);
            } else {
                return binarySearch(num, left, mid - 1, search);
            }
        } else
            return -1;
    }
}

File: Binary Search/RecursiveBinarySearch.py
def binarySearch(num,left,right,search):
    
    mid =(left+right)//2
    
    if left <= right :
        if num[mid] == search :
            return mid
        elif num[mid] < search :
            return binarySearch(num,mid+1,right,search)
        else :
            return binarySearch(num,left,mid-1,search)
    else:
        return -1
    
numbers=[]
print("Enter array size : ",end=" ")
size = int(input())

print("\nEnter sorted array elements : ",end=" ")
numbers=input();
numbers = numbers.split()
numbers =[int(x) for x in numbers]

print("\n Enter the number you wanna search: ",end=" ")   
searchingItem=int(input())

index=binarySearch(numbers,0,len(numbers)-1,searchingItem)

if index==-1 :
    print("\n Item not found")
else :
    print(f'\n Item found at index : {index}')